catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/PublishedEventHandler.java

```diff
@@ -47,89 +54,110 @@ public PublishedEventHandler(Predicate finalPredicate, Subscription subscription
         this.subscription = subscription;
         this.preDelivery = preDelivery;
         this.catalog = catalog;
+        //this will queue tasks to execute 1 task at a time submitting a task is a non blocking call.
+        //the thread is cached but if it is not used for more than 60 seconds it will be destroyed.
```

Comment:
Is there any reason you didn't just create with `Executors.newSingleThreadedExecutor()`? Gives you the same basic semantics, plus or minus a little.